src/modules/sensors/rc_update.cpp

```diff
@@ -228,6 +228,23 @@ RCUpdate::rc_poll(const ParameterHandles &parameter_handles)
 			/* signal is lost or no enough channels */
 			signal_lost = true;
 
+			//This problem occurs at RC controllers which are use PPM signal on RFD 868+/900 Modems (Maybe it can be also related another types)
+
+		} else if (rc_input.channel_count ==
+			   16) { // check failsafe by channel values to detect RC loss (This statement is specific for only RF modules that still sending valid PPM signal when RC loss from ground RF)
+			for (unsigned int i = 0; i < 16; i++) {
+				if (i < 12 && rc_input.values[i] > 999 && rc_input.values[i] < 1005) {
+					signal_lost = true;
+
+				} else if (rc_input.values[i] == 0) {
+					signal_lost = true;
```

Comment:
What I realize now is that if you have any RC receiver with 16 channels where the last channel(s) are 0, this will cause a signal lost even though the other channels might be valid.

@bozkurthan do I see that correctly?

Reply:
Actually, I realize that PX4 logic counts correct channels every time. As far as I know, It is not possible to generate ppm signal with 0 values except RFD 868+/900 modems' problems. If the value of channel is 0 then it shouldn't be count as channel. Or channels couldn't have different min-max value. If 1 channel is between 1000-2000 another one shouldn't be between 0-1000.

Normally, it generates 1000 μs - 2000 μs ( or your arrangement ) with 300 μs pulse width from 8 channels( or your arrangement ). But when problem occurs it generates directly 1000 μs value for first 12 channels  and last 4 channels turns 0. Also, it is not related to channels' value or number of channel The trick is that there is positive pulse width between channels which causes PX4 thinks 16 channels when RC loss. Because PX4 counts pulse width between channels.

From this perspective, I checked all channels in every RC_Update.
The meaning of code:
1)Check the RC has 16 channels.
2)Check first 12 channels' value for between 1000-1005. (Because of RF changes value to this scope)
3)Check last 4 channels' value for 0.
4)Else statement. Circuit breaker.

Actually, answer of your question is in 4. step. If the value of any channel is different from 1000 -1005 (for first 12 channels) or 0 (for last 4 channels) then the failsafe can't be triggered.

Reply:
Oh thanks, I missed the `break`, in that case it's correct :smile:.